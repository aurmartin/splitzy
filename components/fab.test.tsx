import * as React from "react";
import { render, screen, userEvent } from "@testing-library/react-native";
import FAB from "./fab";
import { Text } from "react-native";

describe("FAB", () => {
  it("renders correctly", () => {
    const mockOnPress = jest.fn();
    render(
      <FAB onPress={mockOnPress}>
        <Text>+</Text>
      </FAB>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("handles press correctly", async () => {
    const user = userEvent.setup();

    const mockOnPress = jest.fn();
    render(
      <FAB onPress={mockOnPress}>
        <Text>+</Text>
      </FAB>,
    );

    await user.press(screen.getByTestId("fab-button"));
    expect(mockOnPress).toHaveBeenCalled();
  });
});
