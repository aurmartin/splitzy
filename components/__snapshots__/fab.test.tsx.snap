// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAB renders correctly 1`] = `
<View
  style={
    {
      "bottom": 20,
      "position": "absolute",
      "right": 8,
      "zIndex": 100,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "backgroundColor": "hsl(200 40% 50%)",
          "borderRadius": 16,
          "elevation": 2,
          "padding": 16,
          "shadowColor": "black",
        },
        {
          "overflow": "hidden",
        },
      ]
    }
    testID="fab-button"
  >
    <Text>
      +
    </Text>
  </View>
</View>
`;
